Model/Api/AdyenDonations.php

```diff
@@ -80,10 +80,21 @@ public function donate($payload)
         /** @var Order */
         $order = $this->orderFactory->create()->load($this->checkoutSession->getLastOrderId());
 
+        $this->incrementTryCount($order);
```

Comment:
Shouldn't we be saving this `donationTryCount` only if a donation has failed?

Reply:
Seems to me if a donation has succeeded we can just delete the token end of story

Reply:
If the donation fails, it just throws an exception in the execute method and doesn't return here. So I decided to count at the beginning to be on the safe side for the failing scenarios.

Reply:
https://github.com/Adyen/adyen-magento2/pull/1428/commits/454cf56730d6f05442ed9a0a73d4d8095ee42285